File: src/test/java/blackbox/writer/LineDelimiterTest.java
package blackbox.writer

import net.codinux.csv.kcsv.writer.LineDelimiter
import net.codinux.csv.kcsv.writer.LineDelimiter.Companion.of
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class LineDelimiterTest {
  @Test
  fun test() {
    Assertions.assertEquals("\n", LineDelimiter.LF.toString())
    Assertions.assertEquals("\r", LineDelimiter.CR.toString())
    Assertions.assertEquals("\r\n", LineDelimiter.CRLF.toString())
    Assertions.assertEquals(System.lineSeparator(), LineDelimiter.PLATFORM.toString())
  }

  @Test
  fun testOf() {
    Assertions.assertEquals(LineDelimiter.CRLF, of("\r\n"))
    Assertions.assertEquals(LineDelimiter.LF, of("\n"))
    Assertions.assertEquals(LineDelimiter.CR, of("\r"))
    val e = Assertions.assertThrows(IllegalArgumentException::class.java) { of(";") }
    Assertions.assertEquals("Unknown line delimiter: ;", e.message)
  }
}

File: src/commonTest/kotlin/blackbox/writer/LineDelimiterTest.kt
package blackbox.writer

import net.codinux.csv.writer.LineDelimiter
import net.codinux.csv.writer.LineDelimiter.Companion.of
import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertFailsWith

class LineDelimiterTest {
  @Test
  fun test() {
    assertEquals("\n", LineDelimiter.LF.toString())
    assertEquals("\r", LineDelimiter.CR.toString())
    assertEquals("\r\n", LineDelimiter.CRLF.toString())
//    Assertions.assertEquals(System.lineSeparator(), LineDelimiter.PLATFORM.toString())
  }

  @Test
  fun testOf() {
    assertEquals(LineDelimiter.CRLF, of("\r\n"))
    assertEquals(LineDelimiter.LF, of("\n"))
    assertEquals(LineDelimiter.CR, of("\r"))
    val e = assertFailsWith(IllegalArgumentException::class) { of(";") }
    assertEquals("Unknown line delimiter: ;", e.message)
  }
}